### PR TITLE
Update pin_code validation on FacilityDataset model to allow null

### LINKED ISSUE #10355: FacilityDataset.extra_fields fails validation with null/None pin_code
## Observed behavior

Going through the LOD setup flow, I tried to import a learner from a Facility that had set the pin code to `None` / `null` in JSON, but the sync failed to deserialize.

## Errors and logs
```
INFO     2023-03-30 12:40:43,754 Success! We connected to: http://172.27.64.38:8080/api/public/info/?v=3
INFO     2023-03-30 12:40:45,204 Receiving data (0/13, 0B)
INFO     2023-03-30 12:40:45,210 Receiving data (0/13, 0B)
INFO     2023-03-30 12:40:45,847 Receiving data (13/13, 56.24KB)
INFO     2023-03-30 12:40:45,855 Locally integrating received data
WARNING  2023-03-30 12:40:46,318 Error deserializing instance of FacilityDataset with id 7e5258574c0fad31877c19af6c3d2772: {'extra_fields': ["None does not match type 'string'"]}
```

In the `morango_store`, the serialized `FacilityDataset`
```
"extra_fields": "{\n  \"on_my_own_setup\":false,\n  \"pin_code\":null\n}"
```

## Expected behavior

The pin code should allow to be `None`/`null`, but the `extra_fields` validation does not allow it

## Steps to reproduce

Two instances of Kolibri are required

### Setup
For Instance A
1) Provision a new full-facility Kolibri instance through the setup wizard: Group learning > Full device > Create new learning facility
2) Complete the setup
2) Navigate to Facility > Users
2) Add a new Learner, remember their password
3) Navigate to Facility > Settings
4) Set a PIN for the facility <kbd>Create PIN</kbd>
5) Keep this instance running

For Instance B
1) Provision a new learn-only Kolibri instance, running on a different port than Instance A: Group learning > Learn-only device > Create new learning facility > Import one or more existing user accounts from an existing facility
2) Select Instance A from the 'Select network address' modal
3) Input the Learner credentials you created on Instance A
4) Complete the setup

### Reproduction
1) On Instance A, navigate to Facility > Settings and remove the PIN
2) Restart Instance B, it should automatically sync with Instance A within ~1 minute after restarting
3) On Instance B, navigate to Device, and verify you're not prompted for a PIN
4) Additionally, after successful syncing there should be no log entries in Instance B's logs matching:
```
Error deserializing instance of FacilityDataset with id 7e5258574c0fad31877c19af6c3d2772: {'extra_fields': ["None does not match type 'string'"]}
```

## Context

Kolibri 0.16.0a11

### PR DESCRIPTION
**Pull request by photon0205**
## Summary

- Updated `FacilityDataset.extra_fields` JSON schema to allow null for the `pin_code`
- Added regression test to ensure a null/None `pin_code` does not fail validation when `clean_fields` is called

## References

Resolves #10355 

## Reviewer guidance

…

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [x] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/auth/models.py [hunk 1]
```diff
@@ -246,6 +246,14 @@ def reset_to_default_settings(self, preset=None):
             setattr(self, key, value)
         self.save()
 
+    def validate_pin_code(self):
```
Comment (rtibbles) [inline kolibri/core/auth/models.py]: I think your changes to the JSON schema above are sufficient here - we can allow the pin_code to be null, as that what happens when the pin_code value is unset.

Comment (photon0205) [reply kolibri/core/auth/models.py]: Well I thought so myself, I just added that test as @bjester told me to write a regression unit test which sets the pin_code to None/null, and then calls .clean_fields() on the model

Comment (photon0205) [reply kolibri/core/auth/models.py]: And regarding that validate_pin_code function, should i remove it?

Comment (bjester) [reply kolibri/core/auth/models.py]: @photon0205 Your test `test_null_pin_code` should assert that it does NOT raise a validation error when the pin code is null/None. Before changing the schema, that test should fail without any of these other changes, including to the `clean_fields` method or adding this method. Then once you update the schema like you have, which looks correct, that test should then start passing, thereby preventing a future regression in allowing the pin to be null/None.

**Review by bjester — CHANGES_REQUESTED** [review]
The overridden `clean_fields` method changes and new `validate_pin_code` are unnecessary and should be removed. The `test_null_pin_code` should validate that it does NOT raise the error when the pin is null/None.

File: kolibri/core/auth/models.py (hunk 1 above)
Comment (photon0205) [reply kolibri/core/auth/models.py]: @bjester you can check the latest commit

**Comment by akolson** [comment]
@bjester this is more of a question than a comment on the code. Shouldn't we be adding the test [here](https://github.com/learningequality/kolibri/blob/develop/kolibri/core/auth/test/test_api.py#L1455) while passing `null`?

**Comment by bjester** [comment]
@akolson That test case focuses on the API, whereas the location where @photon0205 put the test here seems more focused on the model / dataset handling itself. The dataset connects all models within the syncing mechanism, and the new test here uses the `clean_fields` method which a sync would invoke, so I think its present location is acceptable.

**Review by bjester — APPROVED** [review]
Thank you @photon0205 for sticking with it and getting it completed! :100: 

On an aside, we have our PR template for good reasons. I've updated your PR description and title to utilize the template and to accurately describe the changes. It has a contributor checklist which helps us all make sure important steps are completed. Please keep this in mind when you open other PRs

**Comment by photon0205** [comment]
> Thank you @photon0205 for sticking with it and getting it completed! 💯
> 
> On an aside, we have our PR template for good reasons. I've updated your PR description and title to utilize the template and to accurately describe the changes. It has a contributor checklist which helps us all make sure important steps are completed. Please keep this in mind when you open other PRs

Thank you for taking the time to update my PR description and title. I appreciate your guidance on how to use the PR template correctly and the reminder to follow the contributor checklist in the future. I'm glad to have your support and look forward to contributing to this project more in the future.

**Pull request merged**